File: 34. Find First and Last Position of Element in Sorted Array-1.py
def lower_bound(nums: List[int], target: int) -> int:
    left = 0
    right = len(nums) - 1
    while left <= right:
        mid = (left + right) // 2
        if nums[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return left


class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        start = lower_bound(nums, target)
        if start == n or nums[start] != target:
            return [-1, -1]
        end = lower_bound(nums, target + 1) - 1
        return [start, end]

File: 34. Find First and Last Position of Element in Sorted Array-1.cpp
class Solution {
public:
  vector<int> searchRange(vector<int> &nums, int target) {
    int n = nums.size();
    int first = -1, last = -1;
    for (int i = 0; i < n; ++i)
      if (nums[i] == target) {
        first = i;
        while (i + 1 < n && nums[i + 1] == target)
          i++;
        last = i;
        break;
      }

    return {first, last};
  }
};